homeassistant/components/godice/sensor.py

```diff
@@ -0,0 +1,168 @@
+"""Sensors for GoDice device."""
+
+import logging
+
+from godice import Color as DiceColor
+
+from homeassistant import const
+from homeassistant.components import persistent_notification
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .dice import ConnectionState, DiceConnectionObserver
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ROLLED_NUMBER_SENSOR_DESCR = SensorEntityDescription(
+    key="rolled_number", state_class=SensorStateClass.MEASUREMENT
+)
+COLOR_SENSOR_DESCR = SensorEntityDescription(
+    key="color",
+    device_class=SensorDeviceClass.ENUM,
+    options=[color.name for color in DiceColor],
+)
+CONNECTION_SENSOR_DESCR = SensorEntityDescription(
+    key="connection",
+    device_class=SensorDeviceClass.ENUM,
+    options=[state.name for state in ConnectionState],
+)
+BATTERY_SENSOR_DESCR = SensorEntityDescription(
+    key="battery_level",
+    device_class=SensorDeviceClass.BATTERY,
+    native_unit_of_measurement=const.PERCENTAGE,
+    state_class=SensorStateClass.MEASUREMENT,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Setups Dice sensors."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    device = data["device"]
+    devinfo = data["device_info"]
```

Comment:
I would use `device_info` instead of `devinfo` to avoid confusion

Reply:
makes sense, renamed